### ISSUE TITLE
Allow blocks to run in parallel while sharing resources of a single container

### ISSUE THREAD

**Issue by samkmetz** [mage-ai/mage-ai#5532]
**Is your feature request related to a problem? Please describe.**

We are currently deployed on ECS, and one thing we have found very frustrating is that to take advantage of running multiple blocks in parallel they must each build their own containers. 

This increases the time it takes for a pipeline to complete quite dramatically and increases compute cost. The only other option is to run them all on the main Mage container without run_pipeline_in_one_process enabled, thus putting strain on the other mage services. 

Making this change allows containerized deployments to fully take advantage of that deployment method while optimizing for both latency and cost efficiency. 

**Describe the solution you'd like**

The functionality that we would like is essentially "run_pipeline_in_one_process" with the pipeline executor setting set to ecs, however, without the limitation that all blocks must now run sequentially. 

In other words, when the pipeline is triggered a container is created and the pipeline is run using the resources of that container while allowing blocks with shared dependencies to still run simultaneously. 

**Describe alternatives you've considered**

The only alternative methods (to my knowledge) are to build a container for every block, or execute all blocks locally on the main container thus removing the benefit of running a containerized system in the first place.

**Comment by messerzen** [comment]
It seems to be the same BUG I've opened here. Right? https://github.com/mage-ai/mage-ai/issues/5518

**Comment by samkmetz** [comment]
Just want to bump this. This would be huge for efficiency and parallel processing.
